packages/game-server/src/shared/entities/items/cloth.ts

```diff
@@ -0,0 +1,25 @@
+import { EntityManager } from "../../../managers/entity-manager";
+import { Entity, Entities } from "../../entities";
+import { Interactive, Positionable } from "../../extensions";
+import { Events } from "../../events";
+import { Player } from "../player";
+
+export class Cloth extends Entity {
+  constructor(entityManager: EntityManager) {
+    super(entityManager, Entities.CLOTH);
+
+    this.extensions = [
+      new Positionable(this),
+      new Interactive(this).init({
+        eventName: Events.INTERACT,
+      }),
+    ];
+
+    this.addEventListener(Events.INTERACT, (evt: CustomEventInit<Player>) => {
+      const player = evt.detail;
```

Comment:
One thing I don't like is how I implemented EventTarget inheritance and you can't really pass (in type-safe way) event arguments

Reply:
maybe I'm missing something, but what is the need for the event emitter?  can't we just pass a callback to new Interactive() and have it run some logic?  we lose typesafety to gain what?

```ts
new Interactive(this, {
  onInteract(player: Player) {}
})
```

Reply:
to be able to serialize extension, send to client and deserialize it here